tools/test.py

```diff
@@ -15,19 +15,32 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
+authstring = os.environ.get('USERPASS')
+if authstring is not None:
+  authstring = authstring + "@"
+else:
+  authstring = ""
+
 class TestMiniKeyValue(unittest.TestCase):
   maxDiff = None
   
   def get_fresh_key(self):
-    return b"http://localhost:3000/swag-" + binascii.hexlify(os.urandom(10))
+    return b"http://"+str.encode(authstring)+b"localhost:3000/swag-" + binascii.hexlify(os.urandom(10))
+
+  # handle 302 manually https://github.com/psf/requests/issues/2949
```

Comment:
Hmm, this concerns me a bit about this whole approach. The idea of mkv is that curl and stuff should just work. Is this unique to requests? What does the HTTP spec say?

Reply:
I think they are being too agressive https://github.com/psf/requests/issues/2949#issuecomment-167664262

I will try to check the HTTP spec

Reply:
btw, this is the CVE they are talking about https://cve.mitre.org/cgi-bin/cvename.cgi?name=2014-1829

Reply:
From what I understand requests shouldn't strip the authorization headers according with the HTTP spec https://tools.ietf.org/html/rfc2617#section-3.3

![image](https://user-images.githubusercontent.com/23288462/71495860-ed76db00-2847-11ea-904e-8016ead82cf4.png)